src/CodeGen_C.cpp

```diff
@@ -1787,6 +1829,44 @@ void CodeGen_C::compile(const LoweredFunc &f) {
 
         // And also the metadata.
         stream << "const struct halide_filter_metadata_t *" << simple_name << "_metadata() HALIDE_FUNCTION_ATTRS;\n";
+
+        // Emit overload(s) that accept Halide::Runtime::Buffers instead of halide_buffer_t
+        // (but only if the function is being emitted in NameMangling::CPlusPlus mode)
+        if (name_mangling == NameMangling::CPlusPlus) {
```

Comment:
You can overload a C-linkage function with a C++ linkage function right? I wonder if we really want this limit.